ompi/mca/coll/libpnbc_osc/pnbc_osc.c

```diff
@@ -435,6 +500,79 @@ static inline int NBC_Start_round(NBC_Handle *handle) {
 
     memcpy (&type, ptr, sizeof (type));
     switch(type) {
+      //TODO
+      case PUT:
+        NBC_DEBUG(5,"  PUT (offset %li) ", offset);
+        NBC_GET_BYTES(ptr,putargs);
+        NBC_DEBUG(5,"*buf: %p, count: %i, type: %p, dest: %i, tag: %i)\n", putargs.buf,
+                  putargs.count, putargs.datatype, putargs.dest, handle->tag);
+        /* get an additional request */
+        handle->req_count++;
+        /* get buffer */
+        if(sendargs.tmpbuf) {
+          buf1=(char*)handle->tmpbuf+(long)sendargs.buf;
+        } else {
+          buf1=(void *)sendargs.buf;
+        }
+#ifdef NBC_TIMING
+        Iput_time -= MPI_Wtime();
+#endif
+        tmp = (MPI_Request *) realloc ((void *) handle->req_array, handle->req_count * sizeof (MPI_Request));
+        if (NULL == tmp) {
+          return OMPI_ERR_OUT_OF_RESOURCE;
+        }
+        
+        handle->req_array = tmp;
+        res = handle->win->w_osc_module->osc_put(buf1, putargs.count, putargs.datatype,
+                                         putargs.dest, 0, putargs.count,
```

Comment:
```suggestion
                                         putargs.target, 0, putargs.count,
```
We should name the "other" MPI process as the target, because MPI uses origin and target (not source and destination) for RMA.

Origin is the MPI process that calls the RMA function and target is the other MPI process - irrespective of which way the data is moving.
The origin for PUT is where the data is moved *from*, but for a GET it is where the data is moved *to*.
The target for a PUT is where the data is moved *to*, but for a GET it is where the data is moved *from*.

Reply:
Understood.